spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/MeanSquaredLogarithmicCriterion.scala

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2016 The BigDL Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.intel.analytics.bigdl.nn
+
+import com.intel.analytics.bigdl.nn.abstractnn.TensorCriterion
+import com.intel.analytics.bigdl.tensor.Tensor
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
+import com.intel.analytics.bigdl.utils.T
+
+import scala.reflect.ClassTag
+
+/**
+ * This method is same as `mean_squared_logarithmic_error` loss in keras.
+ * It calculates:
+ * first_log = K.log(K.clip(y, K.epsilon(), None) + 1.)
+ * second_log = K.log(K.clip(x, K.epsilon(), None) + 1.)
+ * and output K.mean(K.square(first_log - second_log))
+ * Here, the x and y can have or not have a batch.
+ * @param ev$1
+ * @param ev
+ * @tparam T The numeric type in the criterion, usually which are [[Float]] or [[Double]]
+ */
+class MeanSquaredLogarithmicCriterion[T: ClassTag]
+  (implicit ev: TensorNumeric[T]) extends TensorCriterion[T] {
+
+  private val epsilon: T = ev.fromType(1e-07)
+  private val maxValue: T = ev.fromType(Double.MaxValue)
+
+  @transient
+  private var buffer1: Tensor[T] = null // first_log
+  @transient
+  private var buffer2: Tensor[T] = null // second_log
```

Comment:
ditto.

Reply:
criterion does not have clearState(), so I use transient variables.